### PR TITLE
Sync all devices after initial pairing

### LINKED ISSUE #14447: Sync all devices after initial pairing
As a user
I want my data to be transferred after I paired my devices
So that all my devices are synchronized

### Description

When restoring seed phrase from the paired process (@Samyoul is working on it)
after the pairing is successful, we should make sure the data is transferred to the new device, over the established TCP connection (in order to skip waku).

Basically, the effect should be identical as the user has clicked on `Sync all` from the old device to the new device.

Essentially replicating what SyncDevices in status-go does, but not over waku, over this TCP connection, which means creating some protobufs and processing them on the other end as we would as they were received from waku.

### PR DESCRIPTION
**Pull request by qfrank**
Closes [#14447](https://github.com/status-im/status-mobile/issues/14447)

relate [PR](https://github.com/status-im/status-mobile/pull/14514) for status-mobile 

relate [comment](https://github.com/status-im/status-mobile/issues/14447#issuecomment-1332441099) which summarized the basic changes needed in this PR

2nd [PR](https://github.com/status-im/status-go/pull/3047) for the refactor

### CODE REVIEW

**Comment by ghost** [comment]
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://status.im/docs/))?
- [ ] Have you tested changes with mobile?
- [ ] Have you tested changes with desktop?

**Comment by qfrank** [comment]
i'll add some tests later, and any suggestions to the draft would be much appreciate!

File: server/pairing/raw_message_handler.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+	}
+	err = accountDB.SaveAccounts(subAccounts)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// TODO (frank) we need figure out a better way to get node config, e.g. get node config from status-mobile?
+func (s *SyncRawMessageHandler) defaultNodeConfig(keystorePath string, keyUID string) (*params.NodeConfig, error) {
```
Comment (qfrank) [inline server/pairing/raw_message_handler.go]: or maybe we can get it from sender

File: protocol/messenger_sync_raw_messages.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+	response, err := m.saveDataAndPrepareResponse(state)
+	if err != nil {
+		return err
+	}
+	publishMessengerResponse(response)
+	return nil
+}
+
+// TODO (frank) this is a copy implementation of the one in ext/service.go, we should refactor this
+func publishMessengerResponse(response *MessengerResponse) {
```
Comment (qfrank) [inline protocol/messenger_sync_raw_messages.go:202]: what's your opinion for the TODO?

File: server/pairing/sync_device_test.go [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+		KeyUID:       activeAccount.KeyUID,
+		Password:     s.password,
+	}
+	configBytes, err := json.Marshal(config)
+	require.NoError(s.T(), err)
+	err = StartUpPairingClient(clientBackend, cs, string(configBytes))
+	require.NoError(s.T(), err)
+	require.NoError(s.T(), clientBackend.Logout())
+
+	serverBrowserAPI := serverBackend.StatusNode().BrowserService().APIs()[0].Service.(*browsers.API)
```
Comment (qfrank) [inline server/pairing/sync_device_test.go:163]: Currently I haven't found a better way to access browsers.API as it's too nested, if u have better way that would be appreciated

**Comment by qfrank** [comment]
refactor for server_pairing.go r not done yet, but it won't block the main purpose for this PR, i may finish it in a separate PR

File: server/pairing/raw_message_handler.go [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+	err = s.saveSubAccounts(account, password, &syncRawMessage)
+	if err != nil {
+		return err
+	}
+
+	nodeConfig, err := s.defaultNodeConfig(keystorePath, account.KeyUID)
+	if err != nil {
+		return err
+	}
+	err = s.backend.StartNodeWithAccount(*account, password, nodeConfig)
```
Comment (qfrank) [inline server/pairing/raw_message_handler.go]: oh, i should call StartNodeWithAccountAndInitialConfig instead and should get settings from sender via protobuf.SyncRawMessage? if this is the case, what changes should be made to settings and node config? These are the ones I can think of, e.g. InstallationID/DataDir/RootDataDir/KeyStoreDir , is there anything else i missed?

**Comment by qfrank** [comment]
after done these, i was wondering why there isn't a feature just sync the db files and then make some small changes to db to fit or may be we will plan it in the future?

File: server/raw_message_collector.go [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+type RawMessageCollector struct {
+	rawMessages []*common.RawMessage
+}
+
+func (r *RawMessageCollector) Collect(rawMessage *common.RawMessage) {
+	r.rawMessages = append(r.rawMessages, rawMessage)
+}
+
+func (r *RawMessageCollector) GetRawMessages() []*common.RawMessage {
```
Comment (qfrank) [inline server/raw_message_collector.go]: should be getRawMessages, i'll change it later

File: api/utils.go [hunk 6]
```diff
@@ -22,7 +22,9 @@ func RunAsync(f func() error) <-chan error {
 
 // HashMessage calculates the hash of a message to be safely signed by the keycard
 // The hash is calulcated as
-//   keccak256("\x19Ethereum Signed Message:\n"${message length}${message}).
+//
+//	keccak256("\x19Ethereum Signed Message:\n"${message length}${message}).
+//
```
Comment (rasom) [inline api/utils.go]: is this change necessary?

File: eth-node/crypto/crypto.go [hunk 7]
```diff
@@ -202,7 +202,8 @@ func generateSecureRandomData(length int) ([]byte, error) {
 // safely used to calculate a signature from.
 //
 // The hash is calulcated as
-//   keccak256("\x19Ethereum Signed Message:\n"${message length}${message}).
+//
+//	keccak256("\x19Ethereum Signed Message:\n"${message length}${message}).
```
Comment (rasom) [inline eth-node/crypto/crypto.go]: same here and other places when comments are changed

File: api/utils.go (hunk 6 above)
Comment (qfrank) [reply api/utils.go]: i agree, `make lint-fix` did these things, do u have this issue? @rasom
